javalib/src/main/scala/java/util/concurrent/CopyOnWriteArrayList.scala

```diff
@@ -0,0 +1,348 @@
+package java.util
+package concurrent
+
+import java.lang.{reflect => jlr}
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+
+import scala.scalajs._
+
+class CopyOnWriteArrayList[E](private var inner: js.Array[E])
+    extends AbstractCopyOnWriteArrayList[E] with List[E] with RandomAccess
+    with Cloneable with Serializable { self =>
```

Comment:
`self =>` should be on its own line, indented 2 spaces, as soon as the entire class declaration does not fit on one line.